include/commissioner/error.hpp

```diff
@@ -36,11 +36,13 @@
 
 #include <string>
 
+#include <commissioner/defines.hpp>
+
 namespace ot {
 
 namespace commissioner {
 
-enum class Error : int
+enum class OT_COMM_MUST_USE_RESULT Error : int
```

Comment:
Do you happen to know if we use the same `__attribute__((warn_unused_result))` with `enum` and not `enum class`?

Reply:
Annotating class/enum with `war_unused_result` is supported by clang. It has not difference between `enum` and `enum class`. I was trying to apply this to `openthread/openthread`, but decided to try it on this repo first to see how it works and if we like it.